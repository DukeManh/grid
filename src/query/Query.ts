import { IQueryAction, QueryAction } from './QueryAction';

interface IQuery {
  from: (table: string, schema?: string) => IQueryAction;
}

export class Query implements IQuery {
  from(table: string, schema?: string) {
    return new QueryAction({ name: table, schema: schema ?? 'public' });
  }
}
