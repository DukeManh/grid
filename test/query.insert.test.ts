import Query from '../src/query';

const query = new Query();

test('basic insert on table', async () => {
  const res = query
    .from('test_table')
    .insert({
      text_col: 'lorem ipsum',
      json_col: { firstName: 'aaa', lastName: 'bbb' },
      number_array_col: '{1,2,3,4,5}',
      text_array_col: "{'aa', 'bb', 'cc', 'dd'}",
    })
    .toSql();
  expect(res).toMatchSnapshot();
});
test('insert should ignore order and filter', async () => {
  const res = query
    .from('users')
    .insert({ name: 'aaaaa', age: 24 })
    .filter('age', '<', '30')
    .order('name')
    .toSql();
  expect(res).toMatchSnapshot();
});
