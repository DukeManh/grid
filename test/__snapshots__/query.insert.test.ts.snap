// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic insert on table 1`] = `
"
  insert into public.test_table (text_col,json_col,number_array_col,text_array_col) 
  values ('lorem ipsum','{\\"firstName\\":\\"aaa\\",\\"lastName\\":\\"bbb\\"}'::jsonb,'{1,2,3,4,5}','{''aa'', ''bb'', ''cc'', ''dd''}')
  ;"
`;

exports[`insert should ignore order and filter 1`] = `
"
  insert into public.users (name,age) 
  values ('aaaaa',24)
  ;"
`;
